.github/workflows/main.yml

```diff
@@ -0,0 +1,78 @@
+name: Test Build and Call With Fake Register
+
+on: [push]
+
+jobs:
+  ExpectedConnectionReset:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y openssl
+
+      - name: Install Python Dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install requests
+
+      - name: Create Dummy Certificates and Keys
+        run: |
+          mkdir -p dummy-certs
+          openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 -keyout dummy-certs/dummy.key -out dummy-certs/dummy.crt -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+
+      - name: Build Docker Compose
+        run: |
+          docker-compose build
+
+      - name: Prepare Certificates
+        run: |
+          mkdir -p ./factory-registration-ref/data/certs
+          cp dummy-certs/dummy.crt ./factory-registration-ref/data/certs/factory_ca.pem
+          cp dummy-certs/dummy.crt ./factory-registration-ref/data/certs/local-ca.pem
+          cp dummy-certs/dummy.key ./factory-registration-ref/data/certs/local-ca.key
+
+      - name: Run Docker Compose
+        run: |
+          docker-compose up -d
+
+      - name: Execute Script with Expectation of ConnectionResetError
+        run: |
+          set +e
+            output=$(python3 fake-lmp-device-register --registration-url "http://localhost:80/sign" --factory fake-test-sample 2>&1)
+            exit_code=$?
+          set -e
+          if [ $exit_code -ne 0 ]; then
+            echo "Non-zero exit code: $exit_code"
+            if echo "$output" | grep -q 'ConnectionResetError'; then
+              echo "Expected connection reset failure occurred."
```

Comment:
Because it is a draft and it was enough to test what I want to test
The changes in the openssl to ensure that those also work in the linux 
I mean, the changes done in the fake register see:

```
```
2023-11-04 08:18:45,852 - __main__ - INFO - Starting main function with options: Factory=fake-test-sample, SOTA Directory=/home/runner/work/factory-registration-ref/factory-registration-ref/var/sota
2023-11-04 08:18:45,852 - __main__ - INFO - Ensure the sota directory exists in the current directory
2023-11-04 08:18:45,852 - __main__ - INFO - Creating sota directory: /home/runner/work/factory-registration-ref/factory-registration-ref/var/sota
2023-11-04 08:18:45,852 - __main__ - INFO - Create a private key and CSR
2023-11-04 08:18:45,852 - __main__ - INFO - Creating key for UUID: 890da696-109f-4593-8107-9c17c0563525, Factory: fake-test-sample
2023-11-04 08:18:45,852 - __main__ - INFO - Generate a private key into current-dir/var/sota
2023-11-04 08:18:45,860 - __main__ - INFO - Read the key from the file
2023-11-04 08:18:45,860 - __main__ - INFO - Create a temp config for OpenSSL's CSR generation
2023-11-04 08:18:45,860 - __main__ - INFO - Use OpenSSL to generate a CSR using the private key and config
2023-11-04 08:18:45,869 - __main__ - INFO - Let's ensure that temp files are deleted
2023-11-04 08:18:45,869 - __main__ - INFO - Write the private key to the file
2023-11-04 08:18:45,869 - __main__ - INFO - Registering device with data: {'name': '890da696-109f-4593-8107-9c17c0563525', 'uuid': '890da696-109f-4593-8107-9c17c0563525', 'csr': '-----BEGIN CERTIFICATE REQUEST-----\nMIIBPjCB5QIBADBKMS0wKwYDVQQDDCQ4OTBkYTY5Ni0xMDlmLTQ1OTMtODEwNy05\nYzE3YzA1NjM1MjUxGTAXBgNVBAsMEGZha2UtdGVzdC1zYW1wbGUwWTATBgcqhkjO\nPQIBBggqhkjOPQMBBwNCAASVloIIZB+Cn4FFYokgbh/0o9F258FgD5vDx/xc/4y1\nba3/h/yzW5ARlHoT0okKREAvooAxPpHeZzelv0DdyxU3oDkwNwYJKoZIhvcNAQkO\nMSowKDAOBgNVHQ8BAf8EBAMCB4AwFgYDVR0lAQH/BAwwCgYIKwYBBQUHAwIwCgYI\nKoZIzj0EAwIDSAAwRQIgG/rioo2sEZ1ktdxLBmK2BD24EP1dP7AiRXgxlBnvEmkC\nIQDJ9Qt[12](https://github.com/foundriesio/factory-registration-ref/actions/runs/6753716122/job/18360482789#step:10:13)bVK42s37VtmnT6BMgvowuKmRMvXQzTuuwjMgw==\n-----END CERTIFICATE REQUEST-----\n', 'hardware-id': 'intel-corei7-64', 'sota-config-dir': '/home/runner/work/factory-registration-ref/factory-registration-ref/var/sota', 'overrides': {'pacman': {'type': '"ostree+compose_apps"', 'tags': '"master"', 'compose_apps_root': '/home/runner/work/factory-registration-ref/factory-registration-ref/var/sota/compose-apps'}...
2023-11-04 08:[18](https://github.com/foundriesio/factory-registration-ref/actions/runs/6753716122/job/18360482789#step:10:19):45,872 - urllib3.connectionpool - DEBUG - Starting new HTTP connection (1): localhost:80
Tr
```



But yes, we can increase the coverage here if we be able to reach out the docker compose up